sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -287,3 +288,30 @@ case object MsSqlServerDialect extends JdbcDialect {
     case _ => None
   }
 }
+
+/**
+ * :: DeveloperApi ::
+ * Default Apache Derby dialect, mapping real on read
+ * and string/byte/short/boolean/decimal on write.
+ */
+@DeveloperApi
+case object DerbyDialect extends JdbcDialect {
+  override def canHandle(url: String): Boolean = url.startsWith("jdbc:derby")
+  override def getCatalystType(
+      sqlType: Int, typeName: String, size: Int, md: MetadataBuilder): Option[DataType] = {
+    if (sqlType == Types.REAL) Some(FloatType) else None
```

Comment:
Thanks for the review comments, Reynold and Michael. Hopefully, my last commit (e6022ed) addresses your concerns:

o I adjusted the indentation of a method signature, per Reynold's advice.

o I simplified an "if" expression, also per Reynold's advice. However, I have a question about this change (see my comment above on the line in question).

Thanks,
-Rick